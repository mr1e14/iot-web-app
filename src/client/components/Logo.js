import React from "react";
import Grid from "@material-ui/core/Grid";
import logo from "../img/logo.png";

const Logo = props => {
  return (
    <Grid className={props.classes.child} item xs={12}>
      <img src={logo} alt="logo" />
    </Grid>
  );
};

export default Logo;
