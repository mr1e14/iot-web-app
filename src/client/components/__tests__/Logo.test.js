import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Logo from "../Logo";
import Grid from "@material-ui/core/Grid";

const exampleClasses = {
  child: "child"
};

describe("Logo", () => {
  let tree;
  let component;
  beforeEach(() => {
    component = renderer.create(<Logo classes={exampleClasses} />);
    tree = component.toJSON();
  });
  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });
  describe("when renders", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(<Logo classes={exampleClasses} />);
    });
    it("is inside a Grid element", () => {
      expect(wrapper.find(Grid).length).toBe(1);
    });
    it("has an image", () => {
      expect(wrapper.find("img").exists()).toEqual(true);
    });
  });
});
