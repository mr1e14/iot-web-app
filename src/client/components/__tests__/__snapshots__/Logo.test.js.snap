// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 child"
>
  <img
    alt="logo"
    src="test-file-stub"
  />
</div>
`;
